NPE when using SyntheticBeanBuildItem with BeanCreator

### Describe the bug

I'm creating a Quarkus extension and one of the things I need is to use the replacement for AfterBeanDiscovery.addBean().

But I get a NullPointerException when Quarkus tries to instantiate the bean at runtime (in generated code, so no idea what is null)

A simple reproducer of the problem can be found at https://github.com/rdebusscher/quarkus-syntheticbean

### Expected behavior

Ability to instantiate an object myself in code and provide it through the 'CDI injection' system of Quarkus.  No exceptions.

### Actual behavior

When accessing the endpoint where the bean is injected, following NPE is thrown.

```
org.jboss.resteasy.spi.UnhandledException: java.lang.RuntimeException: Error injecting be.rubus.microstream.quarkus.integration.ExampleBean be.rubus.microstream.quarkus.GreetingResource.exampleBean
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:368)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:519)
...
Caused by: java.lang.RuntimeException: Error injecting be.rubus.microstream.quarkus.integration.ExampleBean be.rubus.microstream.quarkus.GreetingResource.exampleBean
        at be.rubus.microstream.quarkus.GreetingResource_Bean.create(Unknown Source)
        at be.rubus.microstream.quarkus.GreetingResource_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:111)
...
Caused by: java.lang.NullPointerException
        at be.rubus.microstream.quarkus.integration.ExampleBean_f7ae08679933a92c66701907ce1ebba818ff26b3_Synthetic_Bean.get(Unknown Source)
        at be.rubus.microstream.quarkus.integration.ExampleBean_f7ae08679933a92c66701907ce1ebba818ff26b3_Synthetic_Bean.get(Unknown Source)
        ... 34 more
```

### How to Reproduce?

See https://github.com/rdebusscher/quarkus-syntheticbean

### Output of `uname -a` or `ver`

Darwin 21.6.0 Darwin Kernel Version 21.6.0: Wed Aug 10 14:25:27 PDT 2022; root:xnu-8020.141.5~2/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "11.0.14.1" 2022-02-08 LTS OpenJDK Runtime Environment Zulu11.54+25-CA (build 11.0.14.1+1-LTS) OpenJDK 64-Bit Server VM Zulu11.54+25-CA (build 11.0.14.1+1-LTS, mixed mode)

### GraalVM version (if different from Java)

/

### Quarkus version or git rev

<quarkus.version>2.11.1.Final</quarkus.version>

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.8.2

### Additional information

_No response_

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java
@@ -582,7 +582,6 @@ public final class Beans {
 
     static void validateBean(BeanInfo bean, List<Throwable> errors, Consumer<BytecodeTransformer> bytecodeTransformerConsumer,
             Set<DotName> classesReceivingNoArgsCtor) {
-
         if (bean.isClassBean()) {
             ClassInfo beanClass = bean.getTarget().get().asClass();
             String classifier = bean.getScope().isNormal() ? "Normal scoped" : null;
@@ -708,6 +707,12 @@ public final class Beans {
                 }
             }
         } else if (bean.isSynthetic()) {
+            // synth beans can accidentally be defined with a non-existing scope, throw exception in such case
+            DotName scopeName = bean.getScope().getDotName();
+            if (bean.getDeployment().getScope(scopeName) == null) {
+                throw new IllegalArgumentException("A synthetic bean " + bean + " was defined with invalid scope annotation - "
+                        + scopeName + ". Please use one of the built-in scopes or a valid, registered custom scope.");
+            }
             // this is for synthetic beans that need to be proxied but their classes don't have no-args constructor
             ClassInfo beanClass = getClassByName(bean.getDeployment().getBeanArchiveIndex(), bean.getBeanClass());
             MethodInfo noArgsConstructor = beanClass.method(Methods.INIT);

--- a/independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SynthBeanWithWrongScopeTest.java
+++ b/independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SynthBeanWithWrongScopeTest.java
@@ -1,0 +1,65 @@
+package io.quarkus.arc.test.buildextension.beans;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.quarkus.arc.BeanCreator;
+import io.quarkus.arc.processor.BeanRegistrar;
+import io.quarkus.arc.test.ArcTestContainer;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.Map;
+import javax.enterprise.context.spi.CreationalContext;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class SynthBeanWithWrongScopeTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder()
+            .beanRegistrars(new TestRegistrar()).shouldFail().build();
+
+    @Test
+    public void testInvalidScopeWasDetectedAtBuildTime() {
+        assertNotNull(container.getFailure());
+        assertTrue(container.getFailure().getMessage().contains("was defined with invalid scope annotation"),
+                container.getFailure().getMessage());
+    }
+
+    // no bean defining annotation; registered synthetically
+    static class BeanifyMe {
+
+    }
+
+    public static class MyBeanCreator implements BeanCreator<BeanifyMe> {
+
+        @Override
+        public BeanifyMe create(CreationalContext<BeanifyMe> creationalContext, Map<String, Object> params) {
+            return new BeanifyMe();
+        }
+    }
+
+    static class TestRegistrar implements BeanRegistrar {
+
+        @Override
+        public void register(RegistrationContext context) {
+            context.configure(BeanifyMe.class).unremovable()
+                    .types(BeanifyMe.class)
+                    // assign annotation that isn't a known scope - this should lead to sensible exception
+                    .scope(NotAScope.class)
+                    .creator(MyBeanCreator.class).done();
+        }
+
+    }
+
+    @Target({ TYPE, METHOD, FIELD, PARAMETER })
+    @Retention(RUNTIME)
+    public @interface NotAScope {
+
+    }
+}

